### DISCUSSION TITLE
将项目放在服务器上执行pnpm run dev或build，自动生成的前置文件.umi内文件不正确

### DISCUSSION THREAD

**Discussion by zzm1122** [umijs/umi#12509] [Q&A] [answered]
使用的是"@umijs/max": "^4.2.11", node版本v18.19.1，
在max setup时报错，
<img width="624" alt="1" src="https://github.com/umijs/umi/assets/45454762/feee77ac-9b81-453f-a9a7-454bbfa0d37f">
后面检查.umi文件，发现插件没有自动生成
<img width="629" alt="2" src="https://github.com/umijs/umi/assets/45454762/8d85ba7c-a0f5-4aed-acd6-3096a52fe1f8">
打开exports.ts文件发现useModel之类的没有添加进去
`// @ts-nocheck
// This file is generated by Umi automatically
// DO NOT CHANGE IT MANUALLY!
// defineApp
export { defineApp } from './core/defineApp'
export type { RuntimeConfig } from './core/defineApp'
// plugins
// plugins types.d.ts
// @umijs/renderer-*
export { createBrowserHistory, createHashHistory, createMemoryHistory, Helmet, HelmetProvider, createSearchParams, generatePath, matchPath, matchRoutes, Navigate, NavLink, Outlet, resolvePath, useLocation, useMatch, useNavigate, useOutlet, useOutletContext, useParams, useResolvedPath, useRoutes, useSearchParams, useAppData, useClientLoaderData, useLoaderData, useRouteProps, useSelectedRoutes, useServerLoaderData, renderClient, __getRoot, Link, useRouteData, __useFetcher, withRouter } from '/data/wwwroot/node/ja-code-template/node_modules/.pnpm/@umijs+renderer-react@4.3.0_react-dom@18.3.1_react@18.3.1/node_modules/@umijs/renderer-react';
export type { History, ClientLoader } from '/data/wwwroot/node/ja-code-template/node_modules/.pnpm/@umijs+renderer-react@4.3.0_react-dom@18.3.1_react@18.3.1/node_modules/@umijs/renderer-react'
// umi/client/client/plugin
export { ApplyPluginsType, PluginManager } from '/data/wwwroot/node/ja-code-template/node_modules/.pnpm/umi@4.3.0_@babel+core@7.24.7_@types+react@18.3.3_eslint@8.35.0_prettier@2.8.8_react-dom@17.0._wzwe7uzalji5sksk4q6effhrqe/node_modules/umi/client/client/plugin.js';
export { history, createHistory } from './core/history';
export { terminal } from './core/terminal';
// react ssr
export const useServerInsertedHTML: Function = () => {};
// test
export { TestBrowser } from './testBrowser';`

**Comment by fz6m** [comment]
如果你要用 `useModel` 这个 model 数据流功能，需要在配置文件里打开 `model: {}`

**Comment by zzm1122** [comment, marked as answer]
问题解决了。.umi.ts文件是隐藏文件，一开始并没有将隐藏文件上传到服务器，导致该问题的。
